tools/github-actions/cascading/src/helpers.ts

```diff
@@ -0,0 +1,81 @@
+import {BaseLogger} from './cascading';
+
+/**
+ * Ensure that the sorting based on git have been done correctly, update the list if there is an issue
+ *
+ * @param branches
+ */
+export function verifyBranches(branches: string[]) {
+  return branches.sort((branchA, branchB) => {
+    const aExtract = branchA.match(/release\/([0-9]+)\.([0-9]+)/);
+    const bExtract = branchB.match(/release\/([0-9]+)\.([0-9]+)/);
+    if (!aExtract || !bExtract) {
+      throw new Error(`Format of branch doesnt match the release pattern ${branches.join(',')}`);
+    }
+    const aMin = aExtract[2];
+    const aMaj = aExtract[1];
+    const bMin = bExtract[2];
+    const bMaj = bExtract[1];
+    if (parseInt(aMaj, 10) > parseInt(bMaj, 10)) {
+      return 1;
+    } else if (parseInt(aMaj, 10) < parseInt(bMaj, 10)) {
+      return -1;
+    }
+    return parseInt(aMin, 10) - parseInt(bMin, 10);
```

Comment:
```suggestion
    const aMaj = parseInt(aExtract[1], 10);
    const bMaj = parseInt(bExtract[1], 10);
    if (aMaj  > bMaj) {
      return 1;
    } else if (aMaj  < bMaj) {
      return -1;
    }
    return parseInt(aExtract[2], 10) - parseInt(bExtract[2], 10);
```

Reply:
I will just fix it with the other one : /**
 * For the branches according to semantic versioning
 *
 * @param branches
 */
export function verifyBranches(branches: string[]) {
  return branches.map(branch => {
    const extract = branch.match(/release\/([0-9]+)\.([0-9]+)/);
    if (!extract) {
      throw new Error(`Format of branch doesnt match the release pattern ${branches.join(',')}`);
    }
    return {
      maj: parseInt(extract[1], 10),
      min: parseInt(extract[2], 10),
      branch
    };
  }).sort((branchObjectA, branchObjectB) => {
    if (branchObjectA.maj > branchObjectB.maj) {
      return 1;
    } else if (branchObjectA.maj < branchObjectB.maj) {
      return -1;
    }
    return branchObjectA.min - branchObjectB.min;
  }).map(branchObject => branchObject.branch);
}